### PR TITLE
Add completing tasks functionality

### PR DESCRIPTION
**Pull request by VolhaVaitulevich**
1. Update index.ts - left only class objects and event listeners;
2. Added modal - without functionality (only 'close modal');
3. Updated Application methods to update the page and sort tasks when they are completed;

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,5 +49,8 @@
   },
   "engines": {
     "node": ">=16.0.0"
+  },
+  "dependencies": {
+    "jquery": "^3.6.2"
```
Comment (siarhei-papkou) [inline package.json:54]: image

File: src/app/application.ts [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    this.renderTasks(this.existingTasks);
+
+    this.renderTasks = this.renderTasks.bind(this);
+    this.completeTask = this.completeTask.bind(this);
+    this.showModal = this.showModal.bind(this);
+  }
+
+  renderTasks(tasks: Array<ITask>): void {
+    //remove all tasks from the page and from local storage
+    localStorage.removeItem('tasks');
```
Comment (siarhei-papkou) [inline src/app/application.ts:20]: Please use our `storage` class

File: src/app/application.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import { MyStorage } from './storage';
+import { ETarget, ITask } from './types';
+
+export class Application {
+  private existingTasks: Array<ITask>;
+  private storage: MyStorage;
+
+  constructor(storage: MyStorage) {
+    this.storage = storage;
```
Comment (siarhei-papkou) [inline src/app/application.ts:9]: https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd

File: src/app/application.ts [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  completeTask(eTarget: ETarget): void {
+    const target = eTarget as HTMLElement;
+    const completedTaskId = Number(target.getAttribute('data_id'));
+
+    //Create arrays to distinguish and sort completed and uncompleted tasks
+    const uncompletedTasks: Array<ITask> = this.existingTasks.filter((item) => item.id !== completedTaskId);
+    const completedTasks: Array<ITask> = this.existingTasks.filter((item) => item.id === completedTaskId);
+    completedTasks.forEach((item) => (item.completed = true));
```
Comment (siarhei-papkou) [inline src/app/application.ts:78]: looks too complex.
Try to use findIndex to update task. and sort with custom coporator.

Comment (VolhaVaitulevich) [reply src/app/application.ts]: Updated in a new commit

**Review by siarhei-papkou — APPROVED** [review]

**Pull request merged**